### PR TITLE
Feat: Change UI of ES error placeholder

### LINKED ISSUE #944: MyData page goes blank if there are no assets
image

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

Closes #944 
Changes:-
- Change UI of Error placeholder for ES
- Change error handling scenarios for the Landing page
- Change error handling scenarios for the explore page

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
- index missing
image
image
image

- connection refuced
image
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Frontend: @shahsank3t, @darth-coder00, @Sachin-chaurasiya

### CODE REVIEW

**Comment by harshach** [comment]
@ShaileshParmar11 This looks great!! Thanks

File: catalog-rest-service/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx [hunk 1]
```diff
@@ -15,63 +15,85 @@
...
-      .then((res: SearchResponse) => {
-        setSearchResult(res);
-        if (isUndefined(entityCounts)) {
-          setEntityCounts(
-            getEntityCountByService(
-              res.data.aggregations?.['sterms#Service']?.buckets
-            )
+    setError('');
+
+    const entityList = [
```
Comment (Sachin-chaurasiya) [inline catalog-rest-service/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx:54]: this should be` entityIndexList`.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
One minor change requested otherwise LGTM.

File: catalog-rest-service/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx (hunk 1 above)
Comment (ShaileshParmar11) [reply catalog-rest-service/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx]: changes made

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**